### ISSUE TITLE
2.7.0 build errors with -Wstrict-prototypes and -Wnullability-completeness

### ISSUE THREAD

**Issue by jamesstout** [OneSignal/OneSignal-iOS-SDK#339]
Just updated and got these errors when -Wstrict-prototypes and -Wnullability-completeness are set.
```
.../Pods/OneSignal/iOS_SDK/OneSignalSDK/Framework/OneSignal.framework/Headers/OneSignal.h:405:36: error: this block declaration is not a prototype [-Werror,-Wstrict-prototypes]
typedef void (^OSEmailSuccessBlock)();
                                   ^
                                    void
.../Pods/OneSignal/iOS_SDK/OneSignalSDK/Framework/OneSignal.framework/Headers/OneSignal.h:78:28: error: pointer is missing a nullability type specifier (_Nonnull, _Nullable, or _Null_unspecified) [-Werror,-Wnullability-completeness]
@property(readonly)NSString* actionID;
                           ^
.../Pods/OneSignal/iOS_SDK/OneSignalSDK/Framework/OneSignal.framework/Headers/OneSignal.h:78:28: note: insert '_Nullable' if the pointer may be null
@property(readonly)NSString* actionID;
                           ^
                             _Nullable
.../Pods/OneSignal/iOS_SDK/OneSignalSDK/Framework/OneSignal.framework/Headers/OneSignal.h:78:28: note: insert '_Nonnull' if the pointer should never be null
@property(readonly)NSString* actionID;
                           ^
                             _Nonnull
2 errors generated.
```
I can see `OSEmailSuccessBlock` is new, not sure why `actionID` has suddenly decided it needs nullability type specifier in my setup as it seems to have been that way for a while. Anyway, just FYI. I started to try to clean these up in the project, but it's difficult for an outsider to decide if something should be nullable or not and I quickly had a mess on my hands.

I didn't want to deactivate the warnings in my entire project so my workaround is:

```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wstrict-prototypes"
#pragma clang diagnostic ignored "-Wnullability-completeness"
#import <OneSignal/OneSignal.h>
#pragma clang diagnostic pop

```

**Comment by Nightsd01** [comment]
This is a warning, not an error. We recently decided that moving forward, we will use nullability flags in any new public methods. Unfortunately, adding nullability specifiers to existing methods would break some people's projects, so we're waiting for the next major release (3.0) to make this change.

In the mean time, using those compiler flags you've mentioned will disable the warnings, which is a good workaround.

**Comment by Nightsd01** [comment]
@jamesstout thank you for reporting this, we've added the compiler flags and released a new update (2.7.1).
